sale_order_general_discount/models/product_product.py

```diff
@@ -6,10 +6,8 @@
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
-    general_discount_apply = fields.Boolean(
-        string="Apply general discount",
-        default=True,
-        required=True,
-        help="If this checkbox is ticked, it means changing general discount on sale order "
-        "will impact sale order lines with this related product.",
+    not_general_discount_apply = fields.Boolean(
```

Comment:
It's not convenient to name negative fields. Better something like `bypass_general_discount`.

Reply:
Now I have  applied this suggested change